File: 프로그래머스/lv2/64065. 튜플/튜플.py
def solution(s):
    array=[]
    s=s[2:-2]
    new=s.split('},{')
    new.sort(key=len)
    
    for i in new:
        ii=i.split(',')
        for j in ii:
            if int(j) not in array:
                array.append(int(j))


    
    return array

File: 프로그래머스/2/64065. 튜플/튜플.js
// 원소의 갯수가 작은 것부터 정렬 후 배열에 없는 값을 추가해감

function solution(s) {
    const answer = [];
// s를 배열에 담기
    
    s= s.slice(1,-1)
    const sets = s.match(/{[^}]+}/g);
    const checkArr= [];
    
    for (let i of sets){
        i = i.slice(1,-1).split(',').map(Number);
            checkArr.push(i)
    }
    
    checkArr.sort((a,b)=>a.length - b.length)
    for (const i of checkArr){
        for (const j of i){
            if (!answer.includes(j)){
                answer.push(j)
            }
        }
    }return answer
}